src/clj/game/core/drawing.clj

```diff
@@ -27,45 +28,56 @@
-  ([state side] (draw state side (make-eid state) 1 nil))
-  ([state side n] (draw state side (make-eid state) n nil))
-  ([state side n args] (draw state side (make-eid state) n args))
+  ([state side eid n] (draw state side eid n nil))
   ([state side eid n {:keys [suppress-event]}]
-   (swap! state update-in [side :register] dissoc :most-recent-drawn) ;clear the most recent draw in case draw prevented
-   (wait-for (trigger-event-simult state side (if (= side :corp) :pre-corp-draw :pre-runner-draw) nil n)
-             (let [active-player (get-in @state [:active-player])
-                   n (+ n (get-in @state [:bonus :draw] 0))
-                   draws-wanted n
-                   draws-after-prevent (if (and (= side active-player) (get-in @state [active-player :register :max-draw]))
-                                         (min n (remaining-draws state side))
-                                         n)
-                   deck-count (count (get-in @state [side :deck]))]
-               (when (and (= side :corp) (> draws-after-prevent deck-count))
-                 (win-decked state))
-               (if (or (and (= side active-player)
-                            (get-in @state [side :register :cannot-draw]))
-                       (not (pos? draws-after-prevent))
-                       (not (pos? deck-count)))
-                 (effect-completed state side eid)
-                 (let [to-draw (take draws-after-prevent (get-in @state [side :deck]))
-                       drawn (doall (for [card to-draw] (move state side card :hand)))]
-                   (swap! state assoc-in [side :register :most-recent-drawn] drawn)
-                   (swap! state update-in [side :register :drawn-this-turn] (fnil #(+ % draws-after-prevent) 0))
-                   (swap! state update-in [:stats side :gain :card] (fnil + 0) n)
-                   (swap! state update-in [:bonus] dissoc :draw)
-                   (if (and (not suppress-event) (pos? deck-count))
-                     (wait-for
-                       (trigger-event-sync state side (if (= side :corp) :corp-draw :runner-draw) draws-after-prevent)
-                       (let [eid (make-result eid drawn)]
-                         (trigger-event-sync state side eid (if (= side :corp) :post-corp-draw :post-runner-draw) draws-after-prevent)))
-                     (effect-completed state side eid))
-                   (when (safe-zero? (remaining-draws state side))
-                     (prevent-draw state side))))
-               (when (< draws-after-prevent draws-wanted)
-                 (let [prevented (- draws-wanted draws-after-prevent)]
-                   (system-msg state (other-side side) (str "prevents "
-                                                            (quantify prevented "card")
-                                                            " from being drawn"))))))))
-
+   (swap! state update-in [side :register] dissoc :currently-drawing)
```

Comment:
Since we are clearing `:currently-drawing` at the start here, I believe issue https://github.com/mtgred/netrunner/issues/5974 is still going to be an issue. I'm wondering if the test you wrote for this scenario is giving a false positive since the only cards being drawn are Hedge Funds.

I'm thinking we should only clear `:currently-drawing` at the end of this function and only if the call to the function is the initial draw. If an additional draw occurs while a draw is resolving those additional cards should be added to the existing `:currently-drawn` set. That's at least how I interpret 8.4.3 

> Abilities that resolve while drawn cards are set aside can move those cards or add new cards to that set.

Reply:
this is a mistake, i meant to leave it at the end.